lib/iris/tests/unit/test_Future.py

```diff
@@ -39,6 +39,9 @@ def test_valid_netcdf_no_unlimited(self):
     def test_valid_clip_latitudes(self):
         future = Future()
         new_value = not future.clip_latitudes
+        with warnings.catch_warnings(record=True) as warn:
+            warnings.simplefilter('always')
+            future.clip_latitudes = new_value
```

Comment:
This test also needs to ensure the warning that's been captured is the correct warning. The code to do this can be found in various places throughout the tests, notably in the `cell_datetime_objects` change (see [L61-69](https://github.com/LukeC92/iris/blob/f3f75550ad7994f0a24552e93122298d8633fef7/lib/iris/tests/unit/test_Future.py#L61-L69)).

Reply:
I wrote a context manager for this, which @pp-mo put in the `iris.tests` module: https://github.com/SciTools/iris/blob/0ce7d5b0117856f9d6e3a2746dae364bad453a80/lib/iris/tests/__init__.py#L542

Reply:
@djkirkham that's rather nice 👍 (as is the shiny new previewer there! ⬆️). @LukeC92 feel free to use that context manager here.